lib/server.go

```diff
@@ -3077,13 +3077,6 @@ func (srv *Server) tryTransitionToFastHotStuffConsensus() {
 		return
 	}
 
-	// If we have at least one sync peer configured but are not connected to any sync peers, then it
-	// means that we are still in the process of connecting to a sync peer. We can exit early and wait
-	// for the network manager to connect to a sync peer.
-	if len(srv.networkManager.connectIps) != 0 && srv.SyncPeer == nil {
```

Comment:
All of the connectIPs may be ineligible to sync from if they have the same tip height as us or if their sync types differ from ours. 

This condition should not gate the transition to the FastHotStuffConsensus.